prow/config/tide.go

```diff
@@ -28,6 +28,21 @@ type Tide struct {
 	// TODO: This will only be possible when we allow specifying orgs. At that
 	//       point, verify the above condition.
 	Queries []TideQuery `json:"queries,omitempty"`
+
+	// A key/value pair of an org/repo as the key and merge method to override
+	// the default method of merge. Valid options are squash, rebase, and merge.
+	MergeType map[string]string `json:"merge_method,omitempty"`
```

Comment:
We should add validation here that a user changing config is not adding a merge method for a `org/repo` that no query exists for. Ideally this validation would also be run as part of the test that loads config so we could catch that in CI before the PR merges and config is synced up.

Reply:
Do you have any pointers on another place in prow that does this so I can get an idea of how you would expect this to work?

Reply:
Here is an example of a config validation test: https://github.com/kubernetes/test-infra/blob/c3f921c120d2d4facb9def7068a349598afeb1c3/prow/config/config_test.go#L97
You can add a function to that file that validates `c.Tide.MergeType`.

Reply:
done. Thanks for the pointer.

Reply:
@cjwagner do we want to mirror what we do in plugin config here and allow for a bare `org` to apply to all repos under it, but allow `org/repo` to overwrite it? Or just keep this as `org/repo -> type`?

Reply:
That would be preferable. Other users of Prow might want to use a non-default merge method for all of their repos.

Just make the `MergeMethod` function check `t.MergeType[org]` before resorting to the default if `t.MergeType[org+"/"+repo]` isn't found.

Reply:
Done